### ISSUE TITLE
text encoding/decoding issues w/ WireType.BINARY

### ISSUE THREAD

**Issue by devinrsmith** [OpenHFT/Chronicle-Wire#39]
The following code outputs `obj2.equals(obj3): false` for `"\uD83E\uDDC0 extra"` with `WireType.BINARY`.

With `"\uD83E\uDDC0"`, it throws a runtime exception.

Both example strings work successfully with `WireType.TEXT` and `WireType.JSON`.

```java
import com.google.common.base.Preconditions;
import net.openhft.chronicle.bytes.Bytes;
import net.openhft.chronicle.core.io.IORuntimeException;
import net.openhft.chronicle.wire.Marshallable;
import net.openhft.chronicle.wire.WireIn;
import net.openhft.chronicle.wire.WireOut;
import net.openhft.chronicle.wire.WireType;

import java.nio.ByteBuffer;

/**
 * Created by dsmith on 11/20/16.
 */
public class MarshallableObj implements Marshallable {
    private final StringBuilder builder = new StringBuilder();

    public void clear() {
        builder.setLength(0);
    }

    public void append(CharSequence cs) {
        builder.append(cs);
    }

    @Override
    public void readMarshallable(WireIn wire) throws IORuntimeException {
        builder.setLength(0);
        Preconditions.checkState(wire.getValueIn().textTo(builder) != null);
    }

    @Override
    public void writeMarshallable(WireOut wire) {
        wire.getValueOut().text(builder);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        MarshallableObj that = (MarshallableObj) o;

        return builder.toString().equals(that.builder.toString());
    }

    @Override
    public int hashCode() {
        return builder.toString().hashCode();
    }

    @Override
    public String toString() {
        return builder.toString();
    }

    public static void main(String[] args) {
        final WireType wireType = WireType.BINARY;
        final String exampleString = "\uD83E\uDDC0 extra";

        final MarshallableObj obj1 = new MarshallableObj();
        final MarshallableObj obj2 = new MarshallableObj();
        final MarshallableObj obj3 = new MarshallableObj();

        obj1.append(exampleString);
        obj2.append(exampleString);

        System.out.println("obj1.equals(obj2): " + obj1.equals(obj2));

        final Bytes<ByteBuffer> bytes = Bytes.elasticByteBuffer();
        obj2.writeMarshallable(wireType.apply(bytes));

        final String output = bytes.toString();
        System.out.println("output: [" + output + "]");

        obj3.readMarshallable(wireType.apply(Bytes.from(output)));

        System.out.println("obj2.equals(obj3): " + obj2.equals(obj3));
    }
}
```

**Comment by peter-lawrey** [comment]
I have converted the class into a unit test which passes.

**Comment by peter-lawrey** [comment]
Closing for now as we haven't been able to reproduce it with the latest version.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.22ea11](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.22ea11), [BOM-2.22ea69](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea69)
